src/main/java/uk/gov/pay/directdebit/mandate/dao/MandateSearchDao.java

```diff
@@ -114,10 +114,10 @@ private SqlStatementAndParameters createSqlQuery(MandateSearchParams params, Str
             sqlParams.put("email", "%" + email + "%");
         });
 
-        params.getExternalMandateState().ifPresent(mandateState -> {
+        if (!params.getInternalStates().isEmpty()) {
```

Comment:
I’m not entirely sure about this. It would mean that if someone searches for a non-existent state, they’d get all payments that match their other conditions, rather than nothing, which is what I’d expect.

Reply:
We validate the `MandateSearchParams` created via the end-point so it is not possible for a client to encounter this scenario (searching for an invalid external state). I don't think `MandateSearchDao` is the right place to further validate `MandateSearchParams` so I'm inclined to leave it as-is. From the Dao's perspective it is going to include any internal states into the WHERE clause when they are provided by the `MandateSearchParams`.

Reply:
Fair enough. If the API validation should stop it then it’s essentially a bug if it ever happens.